app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -274,6 +277,14 @@ public boolean onLoadUri(GeckoSession session, String uri, GeckoSession.Navigati
                         return true;
                     }
 
+                    // Check if we should handle an external link
+                    final Uri urlToURI = Uri.parse(uri);
+                    if (!UrlUtils.isSupportedProtocol(urlToURI.getScheme()) &&
+                            callback != null &&
+                            IntentUtils.handleExternalUri(getContext(), GeckoWebView.this, uri)) {
+                        return true;
```

Comment:
`handleExternalUri()` returns a boolean. We may want to return the result here instead of always `true`?